spring-kafka-test/src/main/java/org/springframework/kafka/test/utils/KafkaTestUtils.java

```diff
@@ -239,6 +242,38 @@ public static OffsetAndMetadata getCurrentOffset(String brokerAddresses, String
 		}
 	}
 
+	/**
+	 * Return the end offsets of the requested topic/partitions
+	 * @param <K> the key type.
+	 * @param <V> the value type.
+	 * @param consumer the consumer.
+	 * @param topic the topic.
+	 * @param partitions the partitions, or null for all partitions.
+	 * @return the map of end offsets.
+	 * @since 2.6.5
+	 */
+	public static <K, V> Map<TopicPartition, Long> getEndOffsets(Consumer<K, V> consumer, String topic,
```

Comment:
I think these generic arguments don't give any value or benefits. 
Probably `Consumer<?, ?>` is what would work for end-users. Even if we need to do some casting internally, it is not going to affect target applications.